cmake/app/boilerplate.cmake

```diff
@@ -341,11 +341,17 @@ foreach(root ${BOARD_ROOT})
       message("Board alias ${BOARD_ALIAS} is hiding the real board of same name")
     endif()
   endif()
-  find_path(BOARD_DIR
-    NAMES ${BOARD}_defconfig
-    PATHS ${root}/boards/*/*
-    NO_DEFAULT_PATH
-    )
+  # Perform a recursive search underneath the ${root}/boards directory to find a
+  # file named ${BOARD}_defconfig. Once found, set the BOARD_DIR variable to the
+  # parent directory and set the BOARD_DIR_ROOT variable so it can be used later. Once
+  # the directory is found, this search does not need to be performed again.
+  # Note: find_path() does not search recursively, and is not able to perform this task.
+  if(NOT BOARD_DIR)
+    file(GLOB_RECURSE tmp_BOARD_DIR ${root}/boards/*/${BOARD}_defconfig)
```

Comment:
curious if this works ok on windows?

Reply:
It appears to build OK on Windows still. I had someone test this change with a nested board definition on windows and it built successfully.